src/main/java/net/coderbot/iris/pipeline/TransformPatcher.java

```diff
@@ -0,0 +1,512 @@
+package net.coderbot.iris.pipeline;
+
+import java.util.Collection;
+import java.util.function.*;
+import java.util.stream.*;
+
+import org.antlr.v4.runtime.Token;
+import org.antlr.v4.runtime.tree.pattern.*;
+import org.apache.logging.log4j.*;
+
+import io.github.douira.glsl_transformer.GLSLParser;
+import io.github.douira.glsl_transformer.GLSLParser.*;
+import io.github.douira.glsl_transformer.ast.StringNode;
+import io.github.douira.glsl_transformer.core.*;
+import io.github.douira.glsl_transformer.core.target.*;
+import io.github.douira.glsl_transformer.print.filter.*;
+import io.github.douira.glsl_transformer.transform.*;
+import io.github.douira.glsl_transformer.tree.ExtendedContext;
+import io.github.douira.glsl_transformer.util.CompatUtil;
+import net.coderbot.iris.IrisLogging;
+import net.coderbot.iris.gbuffer_overrides.matching.InputAvailability;
+import net.coderbot.iris.gl.shader.ShaderType;
+import net.coderbot.iris.pipeline.patcher.AttributeShaderTransformer;
+
+/**
+ * The transform patcher (triforce 2) uses glsl-transformer to do shader
+ * transformation.
+ * 
+ * NOTE: This patcher expects (and ensures) that the string doesn't contain any
+ * (!) preprocessor directives. The only allowed ones are #extension and #pragma
+ * as they are considered "parsed" directives. If any other directive appears in
+ * the string, it will throw.
+ * 
+ * TODO: JCPP has to be configured to remove preprocessor directives entirely
```

Comment:
Is this a blocking issue?

Reply:
I'm not knowledgable of what Iris does prior to patching. It will throw if there are unknown directives like #import, #moj_import or anything other than what is defined in glsl-transformer's GLSL grammar.

Reply:
I have removed this comment since it seems JCPP is configured correctly.